rfc/project-cutover.md

```diff
@@ -0,0 +1,109 @@
+
+There are several artifacts that currently live among the `google.com` projects that need to move to the new project that can't necessarily be simply copied over, including:
+* `halconfig` bucket (owned by `spinnaker-marketplace`) - contains all BOMs and service-specific config for each of those BOMs. Configured in Halyard
+* Each microservice container (owned by `spinnaker-marketplace`) - all containers registered in Google Container Registry (GCR), with each layer being stored in Google Cloud Storage (GCS)
+* `spinaker-artifacts` bucket (owned by `spinnaker-marketplace`) - all released `spin` binaries, Halyard binaries, and `sponnet` library.
+* `spinnakerbot` GKE cluster (owned by `spinnaker-build`) - Github bot for various issue and pull-request management runs here.
+
+### Goals and Non-Goals
+
+Goals:
+* Move all resulting artifacts to the new project with minimal disruption
+
+Non-Goals:
+* Change *how* the artifacts are built or published - just where they are done. including:
+  * Start using Github Package Repo
+  * Use Google Cloud Soure Repo during container builds
+  * Use of Jenkins to orchestrate the release process
+
+## Motivation and Rationale
+
+_What is the current state of the world?_
+* Docker containers are orchestrated from a Jenkins instance in `spinnaker-community` and actually built, stored, and served from the `spinnaker-marketplace` project.
+  * A nightly BOM and each release BOM is generated and stored in the `halconfig` bucket, along with copies of all microservice config files from their respective `halconfig/` directories in each repo.
+
+* Halyard, spin, and sponnet releases are all orchested from the same Jenkins instance and are stored in their respective locations in `spinnaker-artifacts`
+
+* `spinnakerbot` can be [built and deployed](https://github.com/spinnaker/spinnaker/blob/master/spinbot/Makefile#L11) by any developers with Google Container Builder (GCB) priviledges on the `spinnaker-marketplace` project.
+
+_Why is this change being proposed?_
+
+* Transparency in the process, remove Google as the sole owner/operator/bottleneck of the whole thing.
+
+_Who have you identified as early customers of this change?_
+
+* End-users and any new non-Google engineers that will be involved in the expanded nightly build-cop and release-manager duties.
+
+## Timeline
+
+_What is the proposed timeline for the implementation?_
+
+ASAP
+
+## Design
+
+_What exactly are you doing?_
+
+
+### `halconfig` bucket
+
+GCS buckets exist in a global namespace, and cannot be directly transferred between projects.
+
+Two possible approaches:
+1. (Preferred): Copy all data from bucket `halconfig` in `spinnaker-marketplace` to `halconfig-temp` in `spinnaker-community`. Then delete the old bucket and recreate it in the new project.
```

Comment:
Is there an SLA on when a bucket deleted from one project is available for re-creation in another project?  If not, then it seems like we'd be setting ourselves up for an unbounded amount of downtime.

Reply:
I don't think so: https://cloud.google.com/storage/docs/deleting-buckets is the only data I could find on it.